syft/pkg/package.go

```diff
@@ -66,6 +67,15 @@ func (p *Package) merge(other Package) error {
 	if p.PURL == "" {
 		p.PURL = other.PURL
 	}
+
+	if p.MetadataType == JavaMetadataType {
+		virtualPaths := p.Metadata.(JavaMetadata).VirtualPath
+		virtualPaths = append(virtualPaths, other.Metadata.(JavaMetadata).VirtualPath...)
+		if t, ok := p.Metadata.(JavaMetadata); ok {
```

Comment:
If we're doing this check `if t, ok := p.Metadata.(JavaMetadata); ok {` anyway, why not do this first instead of a possible panic here `virtualPaths := p.Metadata.(JavaMetadata).VirtualPath` (albeit very unlikely). And we should probably test `other.Metadata.(JavaMetadata)`, too.